torch/testing/_internal/common_utils.py

```diff
@@ -103,6 +103,7 @@
 except ImportError:
     has_pytest = False
 
+NAVI_ARCH = ("gfx1100", "gfx1101")
```

Comment:
This is not the ideal place to put GPU related things.
`common_cuda.py` is.

You may want to take a similar way with `SM80OrLater` + `@unittest.skipIf(not SM80OrLater, "requires sm80 or later.")`

Reply:
We already use this same logic upstream for MI300 in this file.
https://github.com/pytorch/pytorch/blob/cdfe1bffd16bdd28adbe5518038f68e6ac45de8d/torch/testing/_internal/common_utils.py#L108

I'm not sure how we can numerically split Navi, >=gfx11.* might not hold up forever. Happy to take suggestions.

Reply:
> We already use this same logic upstream for MI300 in this file.

Then we should only add NAVI_ARCH and re-use the `runOnRocmArch` defined later.

Reply:
1. I agree with @xinyazhang's suggestion that this kind of device arch-specific code should reside in common_cuda.py like the CUDA ones. 
2. We should move `MI300_ARCH` there as well. This one should be named NAVI3_ARCH IMO, since other Navi families might have a different behaviour. 
3. That said, @jataylo is right that we cannot unfortunately use a numeric comparison to establish whether a certain feature is supported for a given arch. The next best thing we have is going by device properties, if applicable.
4. runOnRocmArch and skipIfRocmArch should really reuse a common function since their logic only has a difference of a `not`.
5. I can see uses for both functions depending on whether excluding or including is easier for the use-case. In this PR's case, excluding Navi3s seems to be the more straightforward way?

@jataylo Unless this PR is gating something urgently needed, could you address the above refactor in this PR, which we can then upstream?